test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportLongRunningTests.cs

```diff
@@ -0,0 +1,462 @@
+            Uri contentLocation = await _testFhirClient.ExportAsync("Patient/", "_type=Observation");
+            IList<Uri> blobUris = await CheckExportStatus(contentLocation);
+
+            // Download exported data from storage account
+            Dictionary<(string resourceType, string resourceId), Resource> dataFromExport = await DownloadBlobAndParse(blobUris);
+
+            // Download resources from fhir server
+            Uri address = new Uri(_testFhirClient.HttpClient.BaseAddress, "Patient/");
+            Dictionary<(string resourceType, string resourceId), Resource> patientData = await GetResourcesFromFhirServer(address);
+
+            Dictionary<(string resourceType, string resourceId), Resource> compartmentData = new Dictionary<(string resourceType, string resourceId), Resource>();
+            foreach ((string resourceType, string resourceId) key in patientData.Keys)
+            {
+                address = new Uri(_testFhirClient.HttpClient.BaseAddress, "Patient/" + key.resourceId + "/Observation");
+
+                // copies all the new values into the compartment data dictionary
+                (await GetResourcesFromFhirServer(address)).ToList().ForEach(x => compartmentData.TryAdd(x.Key, x.Value));
+            }
+
+            compartmentData.ToList().ForEach(x => patientData.TryAdd(x.Key, x.Value));
+            patientData.Union(compartmentData);
+
+            // Assert both data are equal
+            Assert.True(ValidateDataFromBothSources(compartmentData, dataFromExport));
+        }
+
+        [Fact]
+        public async Task GivenFhirServer_WhenAllDataIsExportedToASpecificContainer_ThenExportedDataIsInTheSpecifiedContianer()
+        {
+            // NOTE: Azure Storage Emulator is required to run these tests locally.
+
+            string testContainer = "test-container";
+
+            // Trigger export request and check for export status
+            Uri contentLocation = await _testFhirClient.ExportAsync(parameters: $"_container={testContainer}");
+            IList<Uri> blobUris = await CheckExportStatus(contentLocation);
+
+            // Download exported data from storage account
+            Dictionary<(string resourceType, string resourceId), Resource> dataFromExport = await DownloadBlobAndParse(blobUris);
+
+            // Download all resources from fhir server
+            Dictionary<(string resourceType, string resourceId), Resource> dataFromFhirServer = await GetResourcesFromFhirServer(_testFhirClient.HttpClient.BaseAddress);
+
+            // Assert both data are equal
+            Assert.True(ValidateDataFromBothSources(dataFromFhirServer, dataFromExport));
+            Assert.True(blobUris.All((url) => url.OriginalString.Contains(testContainer)));
+        }
+
+        private bool ValidateDataFromBothSources(Dictionary<(string resourceType, string resourceId), Resource> dataFromServer, Dictionary<(string resourceType, string resourceId), Resource> dataFromStorageAccount)
```

Comment:
Could we put these helper methods in an ExportTestHelpers file so we don't have duplicate code?

Reply:
Moved common methods to a helper class - `ExportTestHelper`